### ISSUE TITLE
Render reported before complete

### ISSUE THREAD

**Issue by drewdaemon** [elastic/elastic-charts#2124]
**Describe the issue**
The library is reporting a complete render one tick too early.

**To Reproduce**
Add a breakpoint or log statement to the body of the `onRenderChange` callback for a chart.

**Expected behaviour**
`onRenderChange` should only be called with `true` when the chart has been rendered to the canvas. Instead, it is being called a tick before the chart is drawn.

**Screenshots**

https://github.com/elastic/elastic-charts/assets/315764/4fa8b2cf-332b-4deb-9e06-025972ff6163

Look for "visualization took to render after data received" in the above recording ^^. It is logged before the chart is visible.

**Version (please complete the following information):**
 - OS: OSX
 - Browser: Brave, Safari
 - Elastic Charts: 59.1.0

**Additional context**
This behavior throws off our performance measurements, making it look like we're rendering the chart faster than we actually are.

The problem may originate with [this change](https://github.com/elastic/elastic-charts/pull/1730/files#diff-585cd5868f3df217df17da718c85016875d9d239b8ed8b596a5fa7ca3d374cc6L41).

**Errors in browser console**
None

**Comment by nickofthyme** [comment]
Related to changes in #1730

**Comment by markov00** [comment]
As commented here https://github.com/elastic/elastic-charts/pull/1730#issuecomment-1167683408
the original callbacks where wrapped around a RAF because:
>If the user attaches very long sync function to a callback then the rendering phase of the event loop will start only after that sync function has completed, freezing the chart until the end of the callback.
Adding a RAF will push the callback in the queue and runs it after the rendering phase, reducing the freeze effect.

We changed that mechanism to give the consumer the control over how to handle that call, but we can probably go back, at least for the `onRenderChange`

**Comment by markov00** [comment]
@drewdaemon in the meantime can you check this works on lens side, by creating something like that:

```
const onRenderChangeFn = (rendered: boolean) => {
    requestAnimationFrame(() => changeSomethingOnRenderingFunction(rendered));
};

<Settings onRenderChange={onRenderChangeFn}
```

**Comment by drewdaemon** [comment]
>  in the meantime can you check this works on lens side

Thanks @markov00 . This does [resolve the issue](https://github.com/elastic/kibana/pull/163002/files#diff-8282b1418609c26c34a21cafe77c7da9064f898138a144ffe4aa07159fd153efR312-R315). 

> we can probably go back, at least for the onRenderChange

I'm in favor of this. It would be nice to have a fix here in the charts library, since we otherwise have to remember to add this to every expression renderer.

**Comment by nickofthyme** [comment]
Yeah I think that's the best solution, I don't see another way to fix it.
